presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitSource.java

```diff
@@ -199,16 +206,15 @@ public static HiveSplitSource bucketed(
                 compactEffectivePredicate,
                 new PerBucket()
                 {
-                    private final Map<Integer, AsyncQueue<InternalHiveSplit>> queues = new ConcurrentHashMap<>();
+                    private final Map<Integer, BucketedSplits> buckets = new ConcurrentHashMap<>();
                     private final AtomicBoolean finished = new AtomicBoolean();
                     private final boolean preloadSplitsForGroupedExecution = isPreloadSplitsForGroupedExecution(session);
```

Comment:
Why do we need to have it disabled by default? Shouldn't it be on by default?

Reply:
Because we didn't test it yet to understand the latency it might bring. Given we are migrating to use `useRewindableSplitSource`, this is no longer valid but still worth answering :)